http/client.go

```diff
@@ -218,6 +207,16 @@ func (c *Client) BinaryRequest(method, url, token string, reqData *RequestData,
 	if reqData.RespReader != nil {
 		reqData.RespReader = resp.Body
 	} else {
+		if method != "HEAD" && resp.ContentLength != 0 {
+			// Read a small amount of data from the response
+			// body so that the client connection can
+			// be reused.
+			size := resp.ContentLength
+			if size > 1024 || size < 0 {
+				size = 1024
+			}
+			resp.Body.Read(make([]byte, size))
```

Comment:
Don't you have to read all of ContentLength for the connection to be ready for the next response?